### ISSUE TITLE
Not working for new firmware 2017

### ISSUE THREAD

**Issue by perfectflowltd** [daviesian/phantom-decoder#1]
Any chance this script could be updated for the new encryption format? DJILogs/HealthyDrone seem to be doing it fine. cheers

**Comment by daviesian** [comment]
Yes, wouldn't that be nice? It's on my (rather long) list. What are you wanting to use it for?

**Comment by perfectflowltd** [comment]
Hi Ian,

I'm a programmer so having all that information I can't get at is 
driving me nuts! :)

I'm starting to set up drone service business and as part of the 
certification here in the UK we have to log our flight times etc and 
create maintenance records. So having the aircraft ID, flight time and 
battery information is a must for me so I can create email alerts when 
servicing is due on my own system. I just need everything in one place. 
I hate having to writing info down on paper or hack information from 
other sites - too many loops to jump through.

I tried out the dronelogbooks website today, I see they have direct 
access to the DJI cloud account info.
If only DJI would release that api ;)

Nice work, it looks like you got quite far before DJI started encrypting 
the logs.

cheers

David.

**Comment by daviesian** [comment]
I will look into it! Do you happen to know whether the new format is actually encrypted, or just different?

**Comment by perfectflowltd** [comment]
Don't know for sure, I came across a few forum posts stating that

http://www.phantompilots.com/threads/tool-win-offline-txt-flightrecord-to-csv-converter.70428/page-16

I think they have the windows programming running again, I still need to 
try it on my windows box.

I'm just wary of putting effort in when we might see a decent access 
tool from DJI soon, not that I've heard anything...

D.

**Comment by daviesian** [comment]
Agreed! Although it's quite a fun exercise, for some definition of 'fun' ;-)

**Comment by patrick-vigneault-veriforce** [comment]
Any news on this? 

 I'm currently working on a small intern app to store flight records and i'm trying to use the info in those files to populate my database. But the decryption here work 50% of the times because of a buffer to small (I'm not really good in python ...) and when it work I only have the header in my csv.

**Comment by skandragon** [comment]
I'm also interested in using the flight data directly, without requiring access to the DJI site to do any sort of decoding or fetching.

From a quick look at the data from about 20 flights I made, I see the Mavic stores slightly different length data, and the data itself seems somehow scrambled, but the frame type, length, and redundant 0xff frame end byte are all in the clear.

Even then, there is a lot of similarity in the columns of some of the data, so I suspect the "encryption" is a simple XOR with a single byte:

```
00000060  00 00 00 00 0f 50 56 24  41 b4 80 8a c3 1d 1e 2b  |.....PV$A......+|
00000070  5f b6 83 85 c7 1f 1e 27  41 b7 86 86 d9 1c 1d 27  |_......'A......'|
00000080  41 50 81 8d c5 1a 1e 27  4d 86 81 83 c6 0a 1e 23  |AP.....'M......#|
00000090  40 bd 87 80 c5 04 31 2d  41 ac 87 8b c6 12 1f 26  |@.....1-A......&|
000000a0  41 bd 81 87 c7 1e 1e 2f  40 b8 82 8b c7 17 1f 2a  |A....../@......*|
000000b0  42 b4 80 82 c6 7b ff 13  04 20 f0 8b 57 ff 01 37  |B....{... ..W..7|
```
Note all the 0x4X, 0x8X, 0xcX 0x2X items seem to line up.  If this data is scrambled, there's a lot of hints it's not something too tricky.

**Comment by lvauvillier** [comment]
You should have a look here ;)
https://github.com/mikeemoo/dji-log-parser/issues/14

**Comment by sybrenstuvel** [comment]
Apparently [this commit](https://github.com/mikeemoo/dji-log-parser/pull/15/files) shows how to decode the files.

**Comment by Steven22728** [comment]
Hi @daviesian 
We are developing a decoder for all DJI log files as part of a project.
It would be great to get your help.
my email is Steven@hawkuav.co.uk
Kind Regards
Steve
